auditlogger.go

```diff
@@ -0,0 +1,58 @@
+package ladon
+
+import (
+	"log"
+	"os"
+	"strings"
+)
+
+// AuditLogger tracks denied and granted authorizations.
+type AuditLogger interface {
+	LogRejectedAccessRequest(request *Request, pool Policies, deciders Policies)
+	LogGrantedAccessRequest(request *Request, pool Policies, deciders Policies)
+}
+
+// AuditLoggerNoOp is the default AuditLogger, that tracks nothing.
+type AuditLoggerNoOp struct{}
```

Comment:
This doesn't have to be moved but in case you want to move it to `audit_logger_noop`